src/main/java/org/openstreetmap/atlas/checks/validation/tag/UnusualLayerTagsCheck.java

```diff
@@ -109,45 +124,223 @@ public boolean validCheckForObject(final AtlasObject object)
         // mark osm id as flagged
         this.markAsFlagged(object.getOsmIdentifier());
 
-        // Gather all edges from the original way
-        final Set<Edge> osmWay = new OsmWayWalker((Edge) object).collectEdges();
-        // Rule: tunnel edges must have a layer tag value in [-5, -1]
-        if (TunnelTag.isTunnel(object)
-                && (!isTagValueValid || layerTagValue.get() > TUNNEL_LAYER_TAG_MAX_VALUE
-                        || layerTagValue.get() < TUNNEL_LAYER_TAG_MIN_VALUE))
+        if (object.tag(LayerTag.KEY) != null && object.tag(LayerTag.KEY).equals("0"))
         {
-            return Optional.of(createFlag(osmWay, this.getLocalizedInstruction(0)));
+            return Optional.of(this.createFlag(object, LAYER_LEVEL_IS_ZERO));
         }
 
-        // Rule: bridge edges must have no layer tag or a layer tag value in [1, 5]
-        if (BridgeTag.isBridge(object) && (!isTagValueValid || layerTagValue.get() != 0)
-                && (!isTagValueValid || layerTagValue.get() > BRIDGE_LAYER_TAG_MAX_VALUE
-                        || layerTagValue.get() < BRIDGE_LAYER_TAG_MIN_VALUE))
+        else if (!this.landUseNotOnGround(object, isTagValueValid).isEmpty())
+        {
+            return Optional
+                    .of(this.createFlag(object, this.landUseNotOnGround(object, isTagValueValid)));
+        }
+        else if (!this.naturalNotOnGround(object, isTagValueValid).isEmpty())
+        {
+            return Optional
+                    .of(this.createFlag(object, this.naturalNotOnGround(object, isTagValueValid)));
+        }
+        else if (!this.highwayNotOnGround(object, isTagValueValid).isEmpty())
+        {
+            return Optional
+                    .of(this.createFlag(object, this.highwayNotOnGround(object, isTagValueValid)));
+        }
+        else if (!this.waterNotOnGround(object, isTagValueValid).isEmpty())
```

Comment:
Looks like this approach is unnecessarily complicated. Couldn't we just check if the condition is true and then pass the necessary instruction  within the braces? For example, `else if (this.waterNotOnGround(object, isTagValueValid)){return Optional
                    .of(this.createFlag(object, WATERWAY_NOT_ON_GROUND_INSTRUCTION));}` and the waterNotOnGround will look like the follow:
`private String waterNotOnGround(final AtlasObject object, final boolean isTagValueValid)
    {
        final Optional<NaturalTag> natural = NaturalTag.get(object);
        return (isTagValueValid
                && (WaterwayTag.get(object).isPresent()
                        || (natural.isPresent() && natural.get().equals(NaturalTag.WATER)))
                && (!this.objectIsTunnel(object) && !this.objectIsBridge(object)
                        && !this.objectIsCovered(object) && (object.tag(LocationTag.KEY) == null
                                || !object.tag(LocationTag.KEY).equalsIgnoreCase("underground"))));
    }`

Reply:
The above approach will remove the redundant calls to the same method. Also the methods called in the else if condition will look cleaner. Let me know what you think of this.

Reply:
This makes sense! I actually first did this because some methods would return two different instructions (ie, highway above ground and highway below ground), however, since then I combined the cases. I will modify my code to look similar to above.

Reply:
Got it, thanks for the clarification!